# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PContent.ipdl
# Commit: 360b12c6b3e4
# Full Hash: 360b12c6b3e4e74203ba90324580a6c5e149b66b
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2019-07-16 04:07:00
# Regressor Bug: 1563604
# File Overlap Count: 1
# Description:
#   Bug 1563604 - Part 1: Confirm detach calls before dropping refs, r=peterv
#   
#   This ensures that the object is not discarded while there are still in-flight
#   messages from the parent process, and vice-versa, which could cause crashes if a
#   CC was run before any pending messages have arrived.
# ==============================================================================

diff -r 80fff55bee77 -r 360b12c6b3e4 dom/ipc/PContent.ipdl
--- a/dom/ipc/PContent.ipdl	Mon Jul 15 17:19:17 2019 +0000
+++ b/dom/ipc/PContent.ipdl	Mon Jul 15 20:44:50 2019 +0000
@@ -1414,18 +1414,22 @@
     async AttachBrowsingContext(BrowsingContextInitializer aInit);
 
     /**
-     * Remove the synced BrowsingContext 'aContext' from the parent.
+     * Remove the synced BrowsingContext with id 'aContextId' from the parent.
      * DetachBrowsingContext is only needed to be called once for any
      * BrowsingContext, since detaching a node in the BrowsingContext detaches
      * the entire sub-tree rooted at that node. Calling DetachBrowsingContext
      * with an already detached BrowsingContext effectively does nothing. Note
      * that it is not an error to call DetachBrowsingContext on a
-     * BrowsingContext belonging to an already detached subtree. The
-     * 'aMoveToBFCache' paramater controls if detaching a BrowsingContext
-     * should move it to the bfcache allowing it to be re-attached if navigated
-     * to.
+     * BrowsingContext belonging to an already detached subtree.
+     *
+     * As the passed-in context is allowed to already be detached, it is passed
+     * by id, rather than using BrowsingContext's serializer.
+     *
+     * Callers should keep the BrowsingContext alive until this async request is
+     * resolved or rejected, in order to ensure that in-flight messages still
+     * have valid targets.
      */
-    async DetachBrowsingContext(BrowsingContext aContext);
+    async DetachBrowsingContext(uint64_t aContextId) returns (bool unused);
 
     /**
      * Removes all of 'aContext'\'s children, and caches them in the
